npm/webpack-dev-server/src/makeDefaultWebpackConfig.ts

```diff
@@ -41,5 +41,24 @@ export function makeDefaultWebpackConfig (
         template: indexHtmlFile,
       }) as any,
     ],
+  } as any
+
+  if (config.sourceWebpackModulesResult.webpackDevServer.majorVersion === 4) {
+    return {
+      ...finalConfig,
+      devServer: {
+        client: {
+          overlay: false,
+        },
+      },
```

Comment:
```suggestion
      devServer: {
        ...finalConfig.devServer,
        client: {
          ...finalConfig?.devServer?.client
          overlay: false,
        },
      },
```

The way we are merging would overwrite any options specified by the user, so we should do this in a way that preserves these values.

Would also be good to check if we can swap the spread of the client to see if we can preserve the overlay if the user explicitly sets this (most configs should have a null value here and it gets defaulted to true later but not sure on this)

Same for v3 below

Reply:
This is just modifying the default config (defined on line 27) before we've even added anything from the user-specified options, so we should be fine doing this. I did add a couple of tests to make sure that user specified dev server options are honored (except the overlay) here: [`fd175c8` (#21599)](https://github.com/cypress-io/cypress/pull/21599/commits/fd175c863f92b811319ae4792a1ee940cc8a3e2f)